opengrok-indexer/src/main/java/org/opengrok/indexer/history/SSCMRepository.java

```diff
@@ -140,8 +140,12 @@ Executor getHistoryLogExecutor(final File file, String sinceRevision) throws IOE
         } else {
             argv.add(file.getName());
         }
-        if (sinceRevision != null && new Scanner(sinceRevision).hasNextInt()) {
-            argv.add("-v" + (Integer.parseInt(sinceRevision) + 1) + ":" + Integer.MAX_VALUE);
+        if (sinceRevision != null) {
+            try (Scanner scanner = new Scanner(sinceRevision)) {
+                if (scanner.hasNextInt()) {
+                    argv.add("-v" + (Integer.parseInt(sinceRevision) + 1) + ":" + Integer.MAX_VALUE);
```

Comment:
I'd use here Scanner's `nextInt` method:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Scanner.html#nextInt()

Reply:
Then I'd have to handle the exceptions which would make the code a bit more difficult to read, I think.

Reply:
Oh, I see, forgot about the exception, then let's leave it as it is